packages/appsync-modelgen-plugin/src/utils/process-index.ts

```diff
@@ -28,3 +30,14 @@ export const processIndex = (model: CodeGenModel) => {
   const deDupedKeyList = keyList.filter(key => !existingIndexNames.includes(key.arguments.name));
   model.directives.push(...deDupedKeyList);
 };
+
+/*
+ * Accepts a model and field name, and potentially empty list of sortKeyFields to generate a unique index name.
+ * e.g. modelName = Employee, fieldName = manager, sortKeyFields = [level]
+ * will generate a name like employeeByManagerAndLevel.
+ * 
+ * This naming logic is used to generate a default index name for @index directives that don't have a name argument.
```

Comment:
I'd explicitly just link to the code snippet https://github.com/aws-amplify/amplify-category-api/blob/main/packages/amplify-graphql-index-transformer/src/utils.ts#L102-L112

Reply:
Added reference to index directive file.